### PR TITLE
Measure occupancy with large numbers of allowed/denied packets

### PR DESCRIPTION
**Pull request by neiljerram**
This commit adds a scenario where we generate large numbers of packets
between 2 pods, and in alternate cycles those packets are either allowed
or denied.  We measure any associated occupancy impact.

### CODE REVIEW

File: k8sfv/pod.go [hunk 1]
```diff
@@ -123,8 +125,21 @@ func createPod(clientset *kubernetes.Clientset, d deployment, nsName string, spe
...
 		err = netlink.LinkSetNsFd(podIf, int(podNamespace.Fd()))
 		panicIfError(err)
-		err = podNamespace.Do(func(_ ns.NetNS) error {
-			return netlink.LinkSetUp(podIf)
+		err = podNamespace.Do(func(_ ns.NetNS) (err error) {
+			err = netlink.LinkSetUp(podIf)
+			if err != nil {
+				return
+			}
+			err = runCommand("ip", "a", "a", ip+"/32", "dev", veth.PeerName)
```
Comment (fasaxc) [inline k8sfv/pod.go]: I'd prefer if you'd spell out the `"ip", "addr", "add"`.  _I_ know what it means because I'm familiar with that command but it'll be hard to parse for a newbie.

File: k8sfv/pod.go [hunk 2]
```diff
@@ -226,3 +241,31 @@ func cleanupAllPods(clientset *kubernetes.Clientset, nsPrefix string) {
 var zeroGracePeriod int64 = 0
 
 var deleteImmediately = &v1.DeleteOptions{GracePeriodSeconds: &zeroGracePeriod}
+
+func runCommand(command string, args ...string) error {
```
Comment (fasaxc) [inline k8sfv/pod.go]: Could you use `Cmd.CombinedOutput()`, or are you keen to keep stdout/err separate?

File: k8sfv/scale_test.go [hunk 3]
```diff
@@ -160,6 +164,49 @@ var _ = Context("with a k8s clientset", func() {
...
+				}
+				pod1 := createPod(clientset, d, nsName, podSpec{})
+				pod2 := createPod(clientset, d, nsName, podSpec{})
+				for ii := 0; ii < 5; ii++ {
+					// Run an nmap scan between the pods.
+					runNmap(pod1, pod2)
+					sumCalicoDeniedPackets(felixIP)
+
+					// Get Felix to GC and dump heap memory profile.
+					exec.Command("pkill", "-USR1", "calico-felix").Run()
```
Comment (fasaxc) [inline k8sfv/scale_test.go]: This feels ripe for making into a utility function.

File: k8sfv/scale_test.go [hunk 4]
```diff
@@ -160,6 +164,49 @@ var _ = Context("with a k8s clientset", func() {
...
+					time.Sleep(2 * time.Second)
```
Comment (fasaxc) [inline k8sfv/scale_test.go]: Can you poll for the file to be created or something instead of Sleeping?

File: k8sfv/pod.go (hunk 2 above)
Comment (neiljerram) [inline k8sfv/pod.go]: Does that mean I don't need the dance with making my own buffers?

File: k8sfv/scale_test.go (hunk 4 above)
Comment (neiljerram) [inline k8sfv/scale_test.go]: Bit fiddly, as I'd have to have the same logic as Felix for the file name (template + timestamp).

File: k8sfv/pod.go (hunk 2 above)
Comment (neiljerram) [inline k8sfv/pod.go]: Nvm, I can go and look that up.
